src/app/housing/effects/housing.spec.ts

```diff
@@ -77,34 +105,92 @@ describe('HousingEffects', () => {
     executor.queue(ActionFactory.search('Melbourne'));
     effect.search$.subscribe(result => {
       expect(result.payload.length).to.equal(0);
     });
   }));
 
   it('will return a filled result', fakeAsync(() => {
-    (mockHouseService.findHouses as sinon.SinonStub).returns(new BehaviorSubject(mockResponse));
+    (mockHouseService.findHouses as sinon.SinonStub).returns(of(mockResponse));
     executor.queue(ActionFactory.search('Sydney'));
     effect.search$.subscribe(result => {
       expect(result.payload).to.eql(mockResponse);
     });
   }));
 
   it('will return empty result when no search term provided', fakeAsync(() => {
+    const toastMessage = 'Reseting search results.';
+
     executor.queue(ActionFactory.search(''));
     effect.search$.subscribe((result) => {
       expect(result.payload.length).to.equal(0);
+      expect((mockToaster.info as sinon.SinonStub).calledWithExactly(toastMessage)).to.be.true;
+    });
+  }));
+
+  it('will gracefully handle an error that occurred in houseService.findHouses', fakeAsync(() => {
+    (mockHouseService.findHouses as sinon.SinonStub).returns(_throw('ka-BOOM!'));
+    const toastMessage = 'Something went horribly wrong while searching for "let\'s break things!".';
+
+    /* tslint:disable-next-line:quotemark */
+    executor.queue(ActionFactory.search("let's break things!"));
+    effect.search$.subscribe((result) => {
+      expect(result.payload.length).to.equal(0);
+      expect((mockToaster.error as sinon.SinonStub).calledWithExactly(toastMessage)).to.be.true;
+    });
+  }));
+
+  // -- HousingAction.ADD_HOUSE --
+
+  it('will return a different action (from ADD_HOUSE to LIST_HOUSES)', fakeAsync(() => {
+    (mockHouseService.addHouse as sinon.SinonStub).returns(of(newHouse));
+    (mockHouseService.getHouses as sinon.SinonStub).returns(of([...mockResponse, newHouse]));
+
+    executor.queue({ type: HousingAction.ADD_HOUSE });
+    effect.addHouse$.subscribe(result => {
+      expect(result.type).to.equal(HousingAction.LIST_HOUSES.toString());
+    });
+  }));
+
+  it('will return list of House when done adding the new record', fakeAsync(() => {
+    const returnValue: House[] = [...mockResponse, newHouse];
+    const toastMessage = 'The new house has successfully been added!';
+    (mockHouseService.addHouse as sinon.SinonStub).returns(of(newHouse));
+    (mockHouseService.getHouses as sinon.SinonStub).returns(of(returnValue));
+
+    executor.queue(ActionFactory.addHouse(newHouse));
+    effect.addHouse$.subscribe((result) => {
+      expect(result.payload.length).to.equal(returnValue.length);
+      expect((mockHouseService.addHouse as sinon.SinonStub).calledWithExactly(newHouse)).to.be.true;
+      expect((mockToaster.success as sinon.SinonStub).calledWithExactly(toastMessage)).to.be.true;
+      expect((mockHouseService.getHouses as sinon.SinonStub).called).to.be.true;
+      expect((mockToaster.error as sinon.SinonStub).neverCalledWithMatch('.*')).to.be.true;
```

Comment:
Do we need all these asserts? It feels like the test is `will return list of House when done adding the new record` but we're checking to that we calling `addHouse`. Perhaps this is two tests in a describe?

```
describe('when a house is added') {
   it('will return list of House when done adding the new record'){}
   it('will actually add the same newhouse'){}
}
```